behat-tests/features/bootstrap/FeatureContext.php

```diff
@@ -197,6 +225,8 @@ public function createPersonalizedElements(TableNode $elementsTable) {
    * @Then I should see :count for the :type count
    */
   public function assertMenuCount($count, $type) {
+    $this->spinUntilAjaxIsFinished();
```

Comment:
I understand that you are trying to move the Ajax statements out of the features (which I have mixed feelings about but can be swayed).  It feels wrong to be throwing the spin functions throughout the assertions where they aren't necessarily needed.  For example, many things are checked after AJAX finishes processing and if we throw these in throughout the assertions we're running that javascript evaluation many more times than need be.  To really do it right to remove the ajax step I feel like we'd have to write a bunch of extra custom actions that merely wrap a plain form submit/link click, etc. with a spin function as well.  At that point we are using custom functions for everything and it's out of control.

I guess - convince me why this is the best way :-)